devnet-sdk/system/chain.go

```diff
@@ -246,3 +257,34 @@ func chainFromDescriptor(d *descriptors.Chain) (Chain, error) {
 
 	return c, nil
 }
+
+func newL2Chain(chainID string, rpcUrl string, l1Users WalletMap, l2Users WalletMap, chainConfig *params.ChainConfig, l1Addresses AddressMap, l2Addresses AddressMap) *l2Chain {
+	clients := newClientManager()
+	chain := &l2Chain{
+		chain: chain{
+			id:          chainID,
+			rpcUrl:      rpcUrl,
+			users:       l2Users,
+			clients:     clients,
+			node:        newNode(rpcUrl, clients),
+			chainConfig: chainConfig,
+		},
```

Comment:
When we call L2chain's `Addresses()` method, currently It will return nil. `l2Addresses` should be added when initialization. currently it is not used.

Reply:
Good catch!

Reply:
Fixed in 2895f546